src/celeritas/track/SortTracksAction.hh

```diff
@@ -0,0 +1,56 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/track/SortTracksAction.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "celeritas/global/ActionInterface.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sort track by status
+ *
+ * Provides Action interface to `celeritas::detail::partition_tracks_by_status`.
+ * 
+ * TODO: Use init params.init to configure sorting strategy
+ *
+ * \sa celeritas::detail::partition_tracks_by_status
+ */
+class SortTracksAction final : public ExplicitActionInterface
+{
+  public:
+    //! Construct with explicit Id
+    explicit SortTracksAction(ActionId id) : id_(id) {}
+
+    //! Default destructor
+    ~SortTracksAction() = default;
+
+    //! Execute the action with host data
+    void execute(CoreHostRef const& core) const final;
+
+    //! Execute the action with device data
+    void execute(CoreDeviceRef const& core) const final;
+
+    //! ID of the action
+    ActionId action_id() const final { return id_; }
+
+    //! Short name for the action
+    std::string label() const final { return "sort-tracks"; }
+
+    //! Description of the action for user interaction
+    std::string description() const final { return "sort tracks states"; }
+
+    //! Dependency ordering of the action
+    ActionOrder order() const final { return ActionOrder::start; }
```

Comment:
Maybe we should also add an `ActionOrder` data member? And add new values in the action ordering in `celeritas/Types.hh` (later we can change this to something more elegant):
```c++
enum class ActionOrder
{
    sort_start,
    start,  //!< Initialize tracks
    pre,  //!< Pre-step physics and setup
    along,  //!< Along-step
    sort_pre_post,
    pre_post,  //!< Discrete selection kernel
    sort_post,
    post,  //!< After step
    post_post,  //!< User actions after boundary crossing, collision
    sort_end,
    end,  //!< Processing secondaries, including replacing primaries
    size_
};
```